pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,345 @@ import (
+	// MediaEncryption to be supported or not, if yes the algorithms
+	// +nullable
+	MediaEncryption *[]MediaEncryptionAlgorithm `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy to be supported, viz Clear, CryptographicErase
+	// +nullable
+	DataSanitizationPolicy *DataSanitizationPolicy `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// IoConnectivityCapabilities for the pool
+type IoConnectivityCapabilities struct {
+	// AccessProtocols to be supported by the pool, viz NVMe, NVMeOverFabrics
+	AccessProtocols []AccessProtocol `json:"accessProtocols,omitempty"`
+}
+
+// IoPerformanceCapabilities for the pool
+type IoPerformanceCapabilities struct {
+	// AverageIOOperationLatencyMicroseconds to be supported by the pool
+	// +nullable
+	AverageIOOperationLatencyMicroseconds *uint64 `json:"averageIoOperationLatencyMicroseconds,omitempty"`
+
+	// MaxIOOperationsPerSecondPerTerabyte to be supported by the pool
+	// +nullable
+	MaxIOOperationsPerSecondPerTerabyte *uint64 `json:"maxIoOperationsPerSecondPerTerabyte,omitempty"`
+
+	// StorageTier to which the pool belongs to, viz Platinum, Gold, Silver
+	StorageTier StoragePoolTier `json:"storageTier,omitempty"`
+}
+
+// DataProtectionCapabilities for the pool
+type DataProtectionCapabilities struct {
+	// Replication to be supported or not
+	// +nullable
+	Replication *bool `json:"replication,omitempty"`
+
+	// Backup to be supported or not
+	// +nullable
+	Backup *bool `json:"backup,omitempty"`
+
+	// Snapshots to be supported or not
+	// +nullable
+	Snapshots *bool `json:"snapshots,omitempty"`
+}
+
+// AccessMode of the volumes created on pool
+type AccessMode string
+
+const (
+	// ReadWriteOnce can be mounted in read/write mode to exactly 1 host
+	ReadWriteOnce AccessMode = "ReadWriteOnce"
+
+	// ReadOnlyMany can be mounted in read-only mode to many hosts
+	ReadOnlyMany AccessMode = "ReadOnlyMany"
+
+	// ReadWriteMany can be mounted in read/write mode to many hosts
+	ReadWriteMany AccessMode = "ReadWriteMany"
+)
+
+// ProvisioningPolicy for the pool
+type ProvisioningPolicy string
+
+const (
+	// ThickProvisioning for thick provisioning
+	ThickProvisioning ProvisioningPolicy = "thick"
+
+	// ThinProvisioning for thin provisioning
+	ThinProvisioning ProvisioningPolicy = "thin"
+)
+
+// StoragePoolMultipathingOption supported by the pool
+type StoragePoolMultipathingOption string
+
+const (
+	OnlineActiveMultipathingOption  StoragePoolMultipathingOption = "OnlineActive"
+	OnlinePassiveMultipathingOption StoragePoolMultipathingOption = "OnlinePassive"
+	NoneMultipathingOption          StoragePoolMultipathingOption = "None"
+)
+
+// CompressionAlgorithm supported by the pool
+type CompressionAlgorithm string
+
+// MediaEncryptionAlgorithm supported by the pool
+type MediaEncryptionAlgorithm string
+
+// DataSanitizationPolicy supported by the pool
+type DataSanitizationPolicy string
+
+const (
+	NoneDataSanitizationPolicy               DataSanitizationPolicy = "None"
```

Comment:
[suggest]: Naming should first consist of common value i.e. prefix & then the specific value
i.e. `DataSanitizationPolicyNone`, `DataSanitizationPolicyClear`, etc